osf/models/registration_bulk_upload_job.py

```diff
@@ -0,0 +1,54 @@
+from enum import IntEnum
+
+from django.db import models
+
+from osf.models.base import BaseModel
+from osf.utils.fields import NonNaiveDateTimeField
+
+
+class JobState(IntEnum):
+    """Defines the six states of registration bulk upload jobs.
+    """
+
+    PENDING = 0  # Database preparation is in progress
+    INITIALIZED = 1  # Database preparation has been finished and is ready to be picked up
+    PICKED_UP = 2  # Registration creation is in progress
+    DONE_FULL = 3  # All (draft) registrations have been successfully created
+    DONE_PARTIAL = 4  # Some (draft) registrations have failed the creation creation process
+    DONE_ERROR = 5  # All (draft) registrations have failed
+
+
+class RegistrationBulkUploadJob(BaseModel):
+    """Defines a database table that stores registration bulk upload jobs.
+    """
+
+    # The hash of the CSV template payload
+    payload_hash = models.CharField(blank=False, null=False, unique=True, max_length=255)
+
+    # The status/state of the bulk upload
+    state = models.IntegerField(choices=[
+        (JobState.PENDING, 'PENDING'),
```

Comment:
Nitpick: can we do something like this:
```
choices=[
(JobState.PENDING, JobState.PENDING.name),
...
Just to avoid hardcoded strings.
]
```

Reply:
fyi, fixed in https://github.com/CenterForOpenScience/osf.io/pull/9758